test/mann_whitney.jl

```diff
@@ -3,27 +3,144 @@ using HypothesisTests: default_tail
+    @test abs(@inferred(pvalue(ExactMannWhitneyUTest([1:5; ones(5)], [1:10;]); tail = :left)) - 0.0028) <= 1e-4
+
+    # Right tail
+    @test abs(@inferred(pvalue(ExactMannWhitneyUTest([1:10;], [1:10;]); tail = :right)) - 0.5296) <= 1e-4
+    @test abs(@inferred(pvalue(ExactMannWhitneyUTest([1:10;], [2:11;]); tail = :right)) - 0.7634) <= 1e-4
+    @test abs(@inferred(pvalue(ExactMannWhitneyUTest([2:11;], [1:10;]); tail = :right)) - 0.2548) <= 1e-4
+    @test abs(@inferred(pvalue(ExactMannWhitneyUTest([1:10;], [1:5; ones(5)]); tail = :right)) - 0.0028) <= 1e-4
+    @test abs(@inferred(pvalue(ExactMannWhitneyUTest([1:5; ones(5)], [1:10;]); tail = :right)) - 0.9979) <= 1e-4
 end
 
 @testset "Exact with ties and unequal lengths" begin
-    @test abs(@inferred(pvalue(ExactMannWhitneyUTest([1:10;], [2:2:24;]))) - 0.0118) <= 1e-4
-    @test abs(@inferred(pvalue(ExactMannWhitneyUTest([2:2:24;], [1:10;]))) - 0.0118) <= 1e-4
-	show(IOBuffer(), ExactMannWhitneyUTest([1:10;], [2:2:24;]))
+    test1 = ExactMannWhitneyUTest([1:10;], [2:2:24;])
+    test2 = ExactMannWhitneyUTest([2:2:24;], [1:10;])
+    @test test1.nx == test2.ny == 10
+    @test test2.nx == test1.ny == 12
+    @test test1.ranks == test2.ranks[[13:22; 1:12]]
+    @test test1.tie_adjustment == test2.tie_adjustment
+    @test test1.U == 22.5
+    @test test1.U + test2.U == 120
+    @test test1.median == -test2.median
+
+    @test repr(test1) == """
+    Exact Mann-Whitney U test
+    -------------------------
+    Population details:
+        parameter of interest:   Location parameter (pseudomedian)
+        value under h_0:         0
+        point estimate:          -7.5
+
+    Test summary:
+        outcome with 95% confidence: reject h_0
+        two-sided p-value:           0.0120
+
+    Details:
+        number of observations in each group: [10, 12]
+        Mann-Whitney-U statistic:             22.5
+        rank sums:                            [77.5, 175.5]
+        adjustment for ties:                  30.0
+    """
+    @test repr(test2) == """
+    Exact Mann-Whitney U test
+    -------------------------
+    Population details:
+        parameter of interest:   Location parameter (pseudomedian)
+        value under h_0:         0
+        point estimate:          7.5
+
+    Test summary:
+        outcome with 95% confidence: reject h_0
+        two-sided p-value:           0.0120
+
+    Details:
+        number of observations in each group: [12, 10]
+        Mann-Whitney-U statistic:             22.5
+        rank sums:                            [175.5, 77.5]
+        adjustment for ties:                  30.0
+    """
+
+    # Two-sided
+    for kwargs in ((), (; tail = :both))
+        @test abs(@inferred(pvalue(ExactMannWhitneyUTest([1:10;], [2:2:24;]); kwargs...)) - 0.0120) <= 1e-4
+        @test abs(@inferred(pvalue(ExactMannWhitneyUTest([2:2:24;], [1:10;]); kwargs...)) - 0.0120) <= 1e-4
```

Comment:
I'm not sure why but there's a slight discrepancy here, the values reported by R (#337) are `0.01188904`.

Reply:
Can you make an issue about this so it doesn't get lost?
Can't find an immediate source of the discrepancy.